src/main/java/com/cloudogu/scm/review/pullrequest/service/DefaultPullRequestService.java

```diff
@@ -136,6 +137,34 @@ public void reject(Repository repository, PullRequest pullRequest) {
     }
   }
 
+  @Override
+  public boolean hasUserApproved(Repository repository, String pullRequestId, User user) {
+    return getStore(repository).get(pullRequestId).getReviewer().entrySet()
+      .stream()
+      .filter(Map.Entry::getValue)
+      .anyMatch(entry -> entry.getKey().equals(user.getId()));
+  }
+
+  @Override
+  public void approve(Repository repository, String pullRequestId, User user) {
+    PullRequestStore store = getStore(repository);
+    PullRequest pullRequest = store.get(pullRequestId);
+    pullRequest.addApprover(user.getId());
```

Comment:
You could encapsuled duplicate Lines in approve() and disapprove method into new method. Maybe getStoredPullRequest()?

Reply:
That would be two lines, with the following referring back to store and pullRequest. The duplication refers to subscriber which contains the same filter.